The job **`dist-android`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/19802391540/job/56731588989) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/56731588989) [(plain)](https://github.com/rust-lang/rust/commit/3e671f5c808e6624500b9fec4c0ecfdb51541db3/checks/56731588989/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
[RUSTC-TIMING] object test:false 6.063
error[E0061]: this method takes 2 arguments but 3 arguments were supplied
   --> library/std/src/thread/id.rs:75:38
    |
 75 |                 while COUNTER_LOCKED.swap(true, Ordering::Acquire, Ordering::Relaxed) {
    |                                      ^^^^                          ----------------- unexpected argument #3 of type `core::sync::atomic::Ordering`
    |
note: method defined here
   --> /rustc/3e671f5c808e6624500b9fec4c0ecfdb51541db3/library/core/src/sync/atomic.rs:786:12
    |
786 |     pub fn swap(&self, val: bool, order: Ordering) -> bool {
    |            ^^^^
help: remove the extra argument
    |
 75 -                 while COUNTER_LOCKED.swap(true, Ordering::Acquire, Ordering::Relaxed) {
 75 +                 while COUNTER_LOCKED.swap(true, Ordering::Acquire) {
    |

For more information about this error, try `rustc --explain E0061`.
[RUSTC-TIMING] std test:false 3.387
error: could not compile `std` (lib) due to 1 previous error
```

</details>

--- a/library/std/src/thread/id.rs
+++ b/library/std/src/thread/id.rs
@@ -70,7 +70,9 @@ impl ThreadId {
 
                 // Acquire lock.
                 let mut spin = 0;
-                while COUNTER_LOCKED.compare_exchange_weak(false, true, Ordering::Acquire, Ordering::Relaxed).is_err() {
+                // Miri doesn't like it when we yield here as it interferes with deterministically
+                // scheduling threads, so avoid `compare_exchange_weak` to avoid spurious yields.
+                while COUNTER_LOCKED.swap(true, Ordering::Acquire) {
                     if spin <= 3 {
                         for _ in 0..(1 << spin) {
                             spin_loop();
@@ -80,6 +82,7 @@ impl ThreadId {
                     }
                     spin += 1;
                 }
+                // This was `false` before the swap, so we got the lock.
 
                 // SAFETY: we have an exclusive lock on the counter.
                 unsafe {

